lib/lz4file.h

```diff
@@ -0,0 +1,92 @@
+/*
+   LZ4 file library
+   Header File
+   Copyright (C) 2022, Xiaomi Inc.
+   BSD 2-Clause License (http://www.opensource.org/licenses/bsd-license.php)
+
+   Redistribution and use in source and binary forms, with or without
+   modification, are permitted provided that the following conditions are
+   met:
+
+       * Redistributions of source code must retain the above copyright
+   notice, this list of conditions and the following disclaimer.
+       * Redistributions in binary form must reproduce the above
+   copyright notice, this list of conditions and the following disclaimer
+   in the documentation and/or other materials provided with the
+   distribution.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+   A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+   OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+   DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+   THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+   You can contact the author at :
+   - LZ4 source repository : https://github.com/lz4/lz4
+   - LZ4 public forum : https://groups.google.com/forum/#!forum/lz4c
+*/
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+#ifndef LZ4FILE_H
+#define LZ4FILE_H
+
+#include <stdio.h>
+#include "lz4frame.h"
+
+typedef void LZ4F_file;
```

Comment:
Prefer an [incomplete type](https://docs.oracle.com/cd/E19205-01/819-5265/bjals/index.html).
It remains opaque, but it can't be confused with any other pointer type, allowing the compiler to participate in code validation. This is more important than it sounds, as many typical programming errors can be detected this way, saving precious debug time.

In this case, you already have `LZ4_readFile_t` and `LZ4_writeFile_t` defined in `lz4file.c`.
Keep them there, just rename them `struct LZ4_XxxFile_s {... };`,
and here, in the header : `typedef struct LZ4_readFile_s LZ4_readFile_t;`.

The downside is that you have now 2 different types, for read and write operations.
I don't see that as a problem, I actually like this distinction.
But if your goal is to mimic `<stdio.h>`, where the same `FILE*` object is used for all I/O operations, you may want to refactor your type. A simpler adaptation could be to use a `union`, and there are many variations around this theme.
I would, personally, keep them separate, because it matches the API's capabilities, which are solely read and write. You even have separated `LZ4F_readClose()` and `LZ4F_writeClose()`, so not even this operation is shared (while `<stdio.h>` has a single `fclose()`), and there are no additional capability planned. So really no gain  in increasing confusion with a common pointer object between read and write operations.